### PR TITLE
Don't attempt ARMv6 detection on AARCH64.

### PR DESCRIPTION
**Pull request by Dead2**
Fixes compile bug #1616 introduced by #1538

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -102,7 +102,9 @@ set_property(CACHE WITH_SANITIZER PROPERTY STRINGS "Memory" "Address" "Undefined
 if(BASEARCH_ARM_FOUND)
     option(WITH_ACLE "Build with ACLE" ON)
     option(WITH_NEON "Build with NEON intrinsics" ON)
-    option(WITH_ARMV6 "Build with ARMv6 SIMD" ON)
+    if(NOT ARCH STREQUAL "aarch64")
```
Comment (nmoinvaz) [inline CMakeLists.txt]:
```cmake
include(CMakeDependentOption) # put at top
cmake_dependent_option(WITH_ARMV6 "Build with ARMv6 SIMD" ON "NOT ARCH STREQUAL \"aarch64\"" OFF)
```

**Comment by mtl1979** [comment]
I checked output in CMakeError.log and the test for ARMv6 features works correctly, so this patch is incorrect. It's issue with incompatible command-line options. I tested output with just gcc (no extra command-line options) and neither 32-bit or 64-bit compiler enable 32-bit SIMD feature.

Not all compiler targets support `-march=native` correctly even if the compiler accepts it. I tried compiling with `-DWITH_NATIVE_INSTRUCTIONS=ON` using Clang 14 and it did clearly error out saying it's not supported.

I've run the ARMv6 optimized versions on 64-bit ARM and those did actually run without crashing.

I've had weird errors before when trying to compile with clang version that doesn't support the target architecture... Older clang versions could only generate 32-bit binaries for some processor families.

File: configure [hunk 2]
```diff
@@ -1752,20 +1752,22 @@ EOF
...
+                        if test $HAVE_ARMV6_INTRIN -eq 1; then
+                            CFLAGS="${CFLAGS} -DARM_SIMD_INTRIN"
+                            SFLAGS="${SFLAGS} -DARM_SIMD_INTRIN"
+                        fi
 
-                    ARCH_STATIC_OBJS="${ARCH_STATIC_OBJS} slide_hash_armv6.o"
-                    ARCH_SHARED_OBJS="${ARCH_SHARED_OBJS} slide_hash_armv6.lo"
+                        ARCH_STATIC_OBJS="${ARCH_STATIC_OBJS} slide_hash_armv6.o"
+                        ARCH_SHARED_OBJS="${ARCH_SHARED_OBJS} slide_hash_armv6.lo"
+                    fi
```
Comment (mtl1979) [inline configure]: We can make this patch a lot shorter if we just force `$buildarmv6` as `0` when targeting AArch64.

**Review by mtl1979 — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (Dead2) [reply CMakeLists.txt]: Updated 👍

File: configure (hunk 2 above)
Comment (Dead2) [reply configure]: Updated 👍

**Comment by nmoinvaz** [comment]
Honestly I just would prefer it to run the checks on the platform anyways. And we should just fix the checks instead. But if you want this way, the it looks good to me.

**Comment by Dead2** [comment]
@nmoinvaz I don't think it is useful to run checks for things that will never be used, feels a bit like bloat.
But I also think the underlying bug with the tests need to be fixed before this goes in.

**Review by mtl1979 — APPROVED** [review]
LGTM

**Pull request merged**